app/server/utils/i18n.py

```diff
@@ -3,6 +3,7 @@
 
 # kwargs are for placeholders
 def i18n_for(user, key: str, **kwargs) -> str:
-    if user is not None and user.preferred_language is not None:
-        i18n.set('locale', user.preferred_language)
+    preferred_language = user.get('preferred_language') if isinstance(user, dict) else user.preferred_language
```

Comment:
same issue as above

Reply:
huh, how has this worked thus far then? we do have tests that test both translations

Reply:
this seems wrong, user is normally actually a user passed in

Reply:
oh wait i see, tests if is dict